src/FunctionApps/python/IntakePipeline/conversion.py

```diff
@@ -1,18 +1,62 @@
 import logging
 import re
 import requests
+import hl7
 from typing import List, Dict
 
 
-def clean_message(message: str, delimiter: str = "\n") -> str:
-    cleaned_message = re.sub("[\r\n]+", delimiter, message)
+def clean_message(message: str) -> str:
+    parsed_message = None
+    try:
+        parsed_message = hl7.parse(message)
+
+        # MSH-7
+        cleansed_datetime = normalize_hl7_datetime(parsed_message[0][7][0])
+        parsed_message[0][7][0] = cleansed_datetime
+
+    except Exception:
+        logging.exception(
+            "Exception occurred while cleaning message.  "
+            + "Passing through original message."
+        )
+
+        return message
```

Comment:
How robust are the failure conditions of `hl7.parse()`? That seems to be a new imported library so if we're pinning our parsing to that we should understand what it can and can't do. I actually at one point wrote a notebook whose purpose is now long passed but that used Python's hl7 parsing library (should be in `src/notebooks/ELR Profiling Template`). When I coded that up, I remember the docs actually recommending using newlines to delimit messages within a batch, so if you check the code you can see that I had to build up the message list through concatenation. I think I even pulled that straight from an example off their site. Do we need to be doing something like that here?

Reply:
You were absolutely right about newlines.  I was just handling MSH before so I didn't run in to any issues.  I've updated the code to handle all segments I found in our input files, and all date/time fields that are outlined in the HL7 specs.  As a result the newline issue is definitely a blocker, and I've updated the code to handle it.

Reply:
This is resolved above and beyond my concerns. Really really well done here, Spencer!